ui/components/app/edit-gas-popover/edit-gas-popover.component.js

```diff
@@ -187,7 +189,13 @@ export default function EditGasPopover({
         );
         break;
```

Comment:
Hey @segun : sorry for this delayed feedback, but the current update transaction function results in loading spinner being shown on UI till changes are reflected in app state. Will there be same behaviour with these new update methods.

`userFeeLevel` and `gas values` are kind of related, I think it would be better to update them in single method. `await` below is helping in achieving this. But if we can have a single methods at controller level, that will be even  better.

Reply:
There was a comment (a long time ago...I will try and find it) that asked that I remove the spinners.

On 2. Yeah, I can have a method that combine both. But again the crux of the PR is to separate these calls. 

@danjm what do you think about combining these 2?

Reply:
@segun I think combining them is fine. What's most important is the removal of the `updateTransaction` function, which has the ability to update _every_ parameter. But I think it is fine to combine these

Reply:
Also, for edit gas fee popup, spinner is important, as this spinner shows up on confirm transaction page. It is important for this spinner to be visible till estimates are updated.

Reply:
okay, @segun let's re-add the logic for showing the spinner